app/src/main/java/ru/thstdio/aa2020/api/Repository.kt

```diff
@@ -1,51 +1,68 @@
 package ru.thstdio.aa2020.api
 
-import android.content.Context
-import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.delay
-import kotlinx.coroutines.withContext
-import kotlinx.serialization.decodeFromString
-import kotlinx.serialization.json.Json
-import ru.thstdio.aa2020.api.response.*
-import ru.thstdio.aa2020.data.Actor
-import ru.thstdio.aa2020.data.Genre
-import ru.thstdio.aa2020.data.Movie
-
-class Repository(private val appContext: Context) {
-    private val jsonFormat = Json { ignoreUnknownKeys = true }
-
-    suspend fun downloadMovies(): List<Movie> {
-        val moviesJson = loadFromJson<List<JsonMovie>>("data.json", Json::decodeFromString)
-
-        val actors = loadActors().associateBy { it.id }
-        val genres = loadGenres().associateBy { it.id }
-        return moviesJson.map { it.toMovie(actors, genres) }
-    }
+import kotlinx.coroutines.async
+import kotlinx.coroutines.coroutineScope
+import kotlinx.serialization.ExperimentalSerializationApi
+import ru.thstdio.aa2020.api.converter.toActor
+import ru.thstdio.aa2020.api.converter.toCinema
+import ru.thstdio.aa2020.api.converter.toCinemaDetail
+import ru.thstdio.aa2020.api.response.ConfigurationResponse
+import ru.thstdio.aa2020.api.response.Genre
+import ru.thstdio.aa2020.api.service.Service
+import ru.thstdio.aa2020.api.service.TimeDbApi
+import ru.thstdio.aa2020.data.CinemaDetail
+import ru.thstdio.aa2020.data.CinemaListWithTotalPage
 
-    suspend fun downloadMovie(id: Long): Movie =
-        withContext(Dispatchers.IO) {
-            delay(1400)
-            downloadMovies().first { it.id == id }
-        }
 
+@ExperimentalSerializationApi
+class Repository() {
+    private val api: TimeDbApi = Service().createApi()
```

Comment:
calling service constructor?

just move createApi() as private fun out of this class
but inside this file